source/_posts/2017-05-12-openwrt_on_apu3.markdown

```diff
@@ -0,0 +1,503 @@
+[    2.455798] Waiting for root device PARTUUID=6c097903-02...
+[    2.659998] usb 3-1: new high-speed USB device number 2 using ehci-pci
+[    2.666595] usb 4-1: new high-speed USB device number 2 using ehci-pci
+[    2.820863] hub 3-1:1.0: USB hub found
+[    2.824725] hub 4-1:1.0: USB hub found
+[    2.828501] hub 3-1:1.0: 4 ports detected
+[    2.832586] hub 4-1:1.0: 4 ports detected
+[    2.950313] Switched to clocksource tsc
+```
+
+Problem lies here: `[    2.455798] Waiting for root device PARTUUID=6c097903-02...`
+
+## SDHCI controller issue
+
+After short investigation it appears, that we don't have support for the SDHCI
+controller on APU board, so we need to enable it. We need to modify the kernel
+configuration, so we use this command:
+
+```sh
+make kernel_menuconfig
+```
+In the config we need to select those drivers:
+
+* Device Drivers > MMC/SD/SDIO card support:
+    * MMC block device driver
+    * Secure Digital Host Controller Interface support
+    * SDHCI support on PCI bus
+
+Now the system should boot without problems.
+
+## Network problems
+
+After booting the systems it appears we don't have any connectivity (ethernet
+nor wifi). When trying `ifconfig -a` we can see only the `lo` interface.
+
+Let's install some additional packages, which should help us investigate
+
+* Base system > busybox > Customize busybox options > Linux System Utilities:
+    * lspci
+    * lsusb
+* Base system > wireless-tools
+
+When the image is built and system is booted on target we can use `lspci -k` to
+check which devices have kernel modules assigned to them and which don't.
+This `lspci` flavour is pretty poor, compared to usual one, supplied with main
+Linux distributions, but should be enough for our uses.
+
+Among others, we can find these devices (VID:DID), which look interesting:
+
+```
+01:00.0 Class 0200: 8086:1539
+02:00.0 Class 0200: 8086:1539
+03:00.0 Class 0200: 8086:1539
+04:00.0 Class 0280: 168c:003c
+```
+
+According to [this page](http://pci-ids.ucw.cz/read/PC/) we're looking for these
+devices:
+
+* `8086:1539` - this is Intel Ethernet controller (I211 Gigabit Network Connection)
+* `168c:003c` - this is Atheros QCA986x/988x 802.11ac Wireless Network Adapter
+
+We need to find drivers for those. It seems, that Intel is using `CONFIG_IGB`
+kernel option for its driver. Module for Atheros card is in OpenWRT.
+Let's deal firs with ethernet controllers:
+
+```sh
+make kernel_menuconfig
+```
+
+Need to mark this driver:
+
+* Device Drivers > Network device support > Ethernet driver support:
+    * Intel(R) 82575/82576 PCI-Express Gigabit Ethernet support
+
+As for the rest:
+
+```sh
+make menuconfig
+```
+
+First let's mark the driver for our wireless card:
+* Kernel modules > Wireless Drivers:
+    * kmod-ath10k
+
+And also some packages we'll need to set up the access-point:
+* Network:
+    * hostapd
+    * wpa_supplicant
+
+Unfortunately, during my build I got and error. After rerunning `make V=s`
```

Comment:
@mek-x it would be useful to show signature of that error. Typically people look for resources based on error they see.